pkgs/by-name/cm/cmake/000-nixpkgs-disable-bin-prefix.diff

```diff
@@ -0,0 +1,36 @@
+diff --git a/Source/cmSearchPath.cxx b/Source/cmSearchPath.cxx
+index 2b992ef2..b9aed49e 100644
+--- a/Source/cmSearchPath.cxx
++++ b/Source/cmSearchPath.cxx
+@@ -151,6 +151,16 @@ void cmSearchPath::AddSuffixes(const std::vector<std::string>& suffixes)
+   }
+ }
+ 
++static bool IsBinPrefixDisabled(void)
++{
++  enum States { UNDEFINED, DISABLED, ENABLED };
++  static States state = UNDEFINED;
++  if (state == UNDEFINED) {
++    state = cmSystemTools::GetEnvVar("NIXPKGS_CMAKE_IGNORE_PREFIX_PATH_FOR_BIN_SEARCH") != std::nullopt ? DISABLED : ENABLED;
++  }
++  return state == DISABLED;
```

Comment:
I read `bool enabled = ...; return enabled`. Any particular motivation to write this the way it is?

Reply:
not sure i understand. there need to be 3 states. undefined, disabled and enabled. with a boolean there are only two states so it is not possible to know if the call to getenv took place or not.

the reason i didn't just do `getenv("NIXPKGS_...") != NULL` was to just avoid searching the env on every call and keep the change minimal impact. tho admittedly it probably doesn't really matter either way.

Reply:
I mean, `state` is just a local variable here, and `if (state == UNDEFINED)` always goes into the then-branch?

Reply:
> I mean, `state` is just a local variable here, and `if (state == UNDEFINED)` always goes into the then-branch?

it is static so
```
static States state = UNDEFINED;
```
so `UNDEFINED` is only the value on the entrance to the first call to the function.

Reply:
Aaah! I should wear glasses xD